godot-core/src/builtin/file.rs

```diff
@@ -0,0 +1,459 @@
+use crate::{
+    engine::file_access::{CompressionMode, ModeFlags},
+    gen::{central::global::Error, classes::FileAccess},
+    obj::Gd,
+};
+use std::{
+    cmp,
+    io::{BufRead, Error as IoError, ErrorKind, Read, Result as IoResult, Seek, Write},
+};
+
+use super::{GodotString, PackedByteArray, PackedStringArray, Variant};
+
+pub struct GFile {
+    fa: Gd<FileAccess>,
+    buffer: Vec<u8>,
+    buffer_size: usize,
+}
+
+impl From<Gd<FileAccess>> for GFile {
+    fn from(value: Gd<FileAccess>) -> Self {
+        Self {
+            fa: value,
+            buffer: Vec::new(),
+            buffer_size: Self::DEFAULT_BUFFER_SIZE,
+        }
+    }
+}
```

Comment:
This is for case whenever some function or obj would actually need `Gd<FileAccess>` (per our discussion in the issue thread) 
- so they destroy the `GFile` and retrieve the `Gd<FileAccess>` (with Godot cursor still in place!) 
- pass it to another function or obj, 
- retrieve it back in similar manner as from `GFile` (Godot cursor will be in position) 
- then create new `GFile` to continue from whenever the other method set the cursor. 

Alternative is to `get_position()` after other operations on retrieved `Gd<FileAccess>`, closing file, create new `GDFile` with `open_*` and use `seek` to get to the intended spot. 

The key is if after dropping internal `GFile` position tracking we actually still *need* the uniqueness.

Reply:
Then we should design the API differently. Instead of `From`, we could do
```rs
let file_access: Gd<FileAccess> = ...;
let gfile = GFile::from_unique(file_access)?;
```
with:
```rs
fn from_unique(file_access: Gd<FileAccess>): Result<Self, ()> {
    if file_access.get_reference_count() == 1 {
        Ok(Self::from_inner(file_access))
    } else {
        Err(())
    }
}
```

Could also have `try_` overload for consistency, and the one without `try` might panic.

Reply:
@Bromeon I think this mechanism can potentially be useful in the future for other facade implementations to make sure that the reference is unique. Check out the `ReferenceNonUniqueError` in the updated PR and possible naming/organizational suggestions.

Returning `Err(())` was non-favored by clippy, so I went down this rabbit hole when declaring new error type :)